oauth2-server/src/main/java/com/clouway/oauth2/jwt/Jwt.java

```diff
@@ -91,12 +97,102 @@ public ClaimSet(String iss, String scope, String aud, Long exp, Long iat, String
       this.typ = typ;
       this.sub = sub;
       this.prn = sub;
+      this.certId = certId;
+      this.customClaims = customClaims;
+    }
 
+    private ClaimSet(Builder builder) {
+      this.iss = builder.iss;
+      this.scope = builder.scope;
+      this.aud = builder.aud;
+      this.exp = builder.exp;
+      this.iat = builder.iat;
+      this.typ = builder.typ;
+      this.sub = builder.sub;
+      this.prn = builder.prn;
+      this.certId = builder.certId;
+      this.customClaims = builder.customClaims;
+    }
+
+    public static Builder newClaimSet() {
+      return new Builder();
     }
 
     @Override
     public String toString() {
-      return MoreObjects.toStringHelper(this).add("iss", iss).add("scope", scope).add("aud", aud).add("exp", exp).add("iat", iat).add("typ", typ).toString();
+      return MoreObjects.toStringHelper(this).add("iss", iss).add("scope", scope).add("aud", aud).add("exp", exp)
+              .add("iat", iat).add("typ", typ).add("certificate_id", certId).add("custom_claims", customClaims).toString();
+    }
+
+
+    public static final class Builder {
```

Comment:
Builder should go first by convention. 

https://github.com/clouway/teamspace/wiki/Java-Code-Convention